### PR TITLE
Improved readability for messages being logged

### LINKED ISSUE #57152: scheduler cache error messages do not read well
/kind bug

There are several error messages in the scheduler cache that are confusing to read, at least one of these surfaced up to end users via Events. I noticed these two in particular but there may be others:

`pod %v state wasn't initial but get assumed`
`pod %v state wasn't assumed but get forgotten`

Whether or not these should actually be surfacing as events, the messages should be cleaned up.

### PR DESCRIPTION
**Pull request by ravisantoshgudimetla**
**What this PR does / why we need it**:
This improves the readability for messages seen by end-user. /cc @jwforres @bsalamat - For UX
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #57152

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/scheduler/schedulercache/cache.go [hunk 1]
```diff
@@ -178,7 +178,7 @@ func (cache *schedulerCache) ForgetPod(pod *v1.Pod) error {
 
 	currState, ok := cache.podStates[key]
 	if ok && currState.pod.Spec.NodeName != pod.Spec.NodeName {
-		return fmt.Errorf("pod %v state was assumed on a different node", key)
+		return fmt.Errorf("pod %v was assumed on %v but is on %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/cache.go]: s/but is on/but assigned to/

File: pkg/scheduler/schedulercache/cache.go [hunk 2]
```diff
@@ -241,7 +241,7 @@ func (cache *schedulerCache) AddPod(pod *v1.Pod) error {
 	case ok && cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
 			// The pod was added to a different node than it was assumed to.
-			glog.Warningf("Pod %v assumed to a different node than added to.", key)
+			glog.Warningf("pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/cache.go]: start with big letter for log

File: pkg/scheduler/schedulercache/cache.go [hunk 3]
```diff
@@ -277,14 +277,14 @@ func (cache *schedulerCache) UpdatePod(oldPod, newPod *v1.Pod) error {
 	// before Update event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != newPod.Spec.NodeName {
-			glog.Errorf("Pod %v updated on a different node than previously added to.", key)
+			glog.Errorf("pod %v updated on a different node than previously added to.", key)
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/cache.go]: start with big letter for log

File: pkg/scheduler/schedulercache/cache.go [hunk 4]
```diff
@@ -304,7 +304,7 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 	// before Remove event, in which case the state would change from Assumed to Added.
 	case ok && !cache.assumedPods[key]:
 		if currState.pod.Spec.NodeName != pod.Spec.NodeName {
-			glog.Errorf("Pod %v removed from a different node than previously added to.", key)
+			glog.Errorf("pod %v was assumed to be on %v but got added to %v", key, pod.Spec.NodeName, currState.pod.Spec.NodeName)
```
Comment (k82cn) [inline pkg/scheduler/schedulercache/cache.go]: ditto

File: pkg/scheduler/schedulercache/cache.go [hunk 5]
```diff
@@ -313,7 +313,7 @@ func (cache *schedulerCache) RemovePod(pod *v1.Pod) error {
 		}
 		delete(cache.podStates, key)
 	default:
-		return fmt.Errorf("pod state wasn't added but get removed. Pod key: %v", key)
+		return fmt.Errorf("pod %v wasn't added so cannot be removed", key)
```
Comment (bsalamat) [inline pkg/scheduler/schedulercache/cache.go]: Please mention cache here:
"pod %v is not found in the scheduler cache, so cannot be removed from it"

File: pkg/scheduler/schedulercache/cache.go (hunk 3 above)
Comment (resouer) [reply pkg/scheduler/schedulercache/cache.go]: You may want to say upper case, haha

**Comment by ravisantoshgudimetla** [comment]
@k82cn @bsalamat I have addressed your comments. PTAL.

File: pkg/scheduler/schedulercache/cache.go [hunk 6]
```diff
@@ -131,7 +131,7 @@ func (cache *schedulerCache) AssumePod(pod *v1.Pod) error {
 	cache.mu.Lock()
 	defer cache.mu.Unlock()
 	if _, ok := cache.podStates[key]; ok {
-		return fmt.Errorf("pod %v state wasn't initial but get assumed", key)
+		return fmt.Errorf("Pod %v's state wasn't initialized, so can't be assumed", key)
```
Comment (bsalamat) [inline pkg/scheduler/schedulercache/cache.go]: I would change this to "Pod %v is not found in the cache, so can't be assumed".

Comment (ravisantoshgudimetla) [reply pkg/scheduler/schedulercache/cache.go]: Done @bsalamat

**Comment by bsalamat** [comment]
/lgtm
Thanks, @ravisantoshgudimetla!

File: pkg/scheduler/schedulercache/cache.go (hunk 6 above)
Comment (k82cn) [reply pkg/scheduler/schedulercache/cache.go]: we should not return error message started with big letter :)

**Comment by k82cn** [comment]
golint would like 1. start with big letter for log message, and 2. start with litter letter for message of error.

**Comment by k82cn** [comment]
no other comments.

File: pkg/scheduler/schedulercache/cache.go [hunk 7]
```diff
@@ -345,7 +345,7 @@ func (cache *schedulerCache) GetPod(pod *v1.Pod) (*v1.Pod, error) {
 
 	podState, ok := cache.podStates[key]
 	if !ok {
-		return nil, fmt.Errorf("pod %v does not exist", key)
+		return nil, fmt.Errorf("Pod %v does not exist in scheduler cache", key)
```
Comment (resouer) [inline pkg/scheduler/schedulercache/cache.go]: As k82cn suggested before, error message should start with lower case.

**Comment by bsalamat** [comment]
/lgtm cancel
Ravi, could you please fix the character cases as Klaus suggested? Thanks!

File: pkg/scheduler/schedulercache/cache.go (hunk 3 above)
Comment (ravisantoshgudimetla) [reply pkg/scheduler/schedulercache/cache.go]: Just to be clear, @k82cn, this is right, it should start with lower case?

**Comment by ravisantoshgudimetla** [comment]
Done @k82cn @bsalamat @resouer - Can someone point me to documentation on how add packages for golint validations? Can taking pkg/scheduler/schedulercache out of hack/.golint_failures help?

**Comment by bsalamat** [comment]
I think removing the package from pkg/scheduler/schedulercache out of hack/.golint_failures does solve the problem, but the package may fail golint for other reasons beyond the scope of this PR.

File: pkg/scheduler/schedulercache/cache.go (hunk 4 above)
Comment (bsalamat) [reply pkg/scheduler/schedulercache/cache.go]: This should be upper case.

**Comment by ravisantoshgudimetla** [comment]
Yes, I created an issue for golint validation in scheduler pkg. #58234

**Comment by bsalamat** [comment]
/lgtm

Thanks a lot, @ravisantoshgudimetla for this PR and also for filing the issue to clean up golint warnings.

**Pull request merged**